### PR TITLE
Editorial mdlint and vale updates for docs/Reference directory

### PR DESCRIPTION
**Pull request by grantnoble**
Line length, sentence case for headings, EOL spaces, passive voice to active voice, other misc. editorial changes.

### CODE REVIEW

**Comment by grantnoble** [comment]
There are quite a few instances of the Markdown lint warming "MD036 Emphasis used instead of a header". Where we use \*\*Parameters** and \*\*Returns**, for example.

**Comment by NicolasMassart** [comment]
> There are quite a few instances of the Markdown lint warming "MD036 Emphasis used instead of a header". Where we use **Parameters** and **Returns**, for example.

I think we should switch to headers indeed. It's easier to link and navigate.

**Comment by NicolasMassart** [comment]
Trailing spaces may also be simple to fix and would remove a lot of instance of this issue

**Comment by grantnoble** [comment]
Agree about the headers. I will update.

File: docs/Reference/API-Methods.md [hunk 1]
```diff
@@ -4256,50 +4490,62 @@ per transaction in the order the transactions were executed.
...
 
 !!! important
-    For production systems requiring private transactions, we recommend using a network
-    with a consensus mechanism supporting transaction finality. For example, [IBFT 2.0](../HowTo/Configure/Consensus-Protocols/IBFT.md).
-    
-    Using private transactions with [pruning](../Concepts/Pruning.md) or [fast sync](CLI/CLI-Syntax.md#sync-mode)
-    is not supported.
 
-    Besu does not implement [`eea_sendTransaction`](../HowTo/Send-Transactions/Account-Management.md).
+    For production systems requiring private transactions, we recommend using a network with a
```
Comment (NicolasMassart) [inline docs/Reference/API-Methods.md]: According to https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/w/we we should try another way write this sentence as here it doesn't explain the benefits of doing so.
So perhaps "For production systems requiring private transactions, use a network with a consensus mechanism supporting transaction finality to make sure the private state can not get inconsistant with the chain. For example, IBFT2 provides the required finality."

Comment (grantnoble) [reply docs/Reference/API-Methods.md]: Reworded slightly, but updated accordingly.

**Pull request merged**